### ISSUE TITLE
Registering collection results in errors in logs 

### ISSUE THREAD

**Issue by mstfdkmn** [irods/irods#7397]
### iRODS Version, OS and Version
4.3.0/4.3.1, almalinux8, iCommands4.3.0/1, prc

### What did you try to do?
I tried to register a directory.

### Expected behavior
I expected to register a directory as a collection in irods without seeing any error in logs.

### Observed behavior (including steps to reproduce, if applicable)
A directory with its content has been registered as a 'good' collection/objects but this operation writes error in logs.

```
[root@dev-u0137480 rods]# ls -al dir_to_register
total 44
drwxrwxrwx 2 irods irods 4096 Oct 27 16:13 .
drwxr-xr-x 5 irods irods 4096 Oct 27 16:08 ..
-rwxrwxrwx 1 irods irods    6 Oct 27 16:13 File000.txt
-rwxrwxrwx 1 irods irods    6 Oct 27 16:13 File001.txt

[irods@dev-u0137480 ~]$ ireg -r /vault/home/rods/dir_to_register /u0137480/home/rods/dir_to_register`
[irods@dev-u0137480 ~]$ ils -L /u0137480/home/rods/dir_to_register
/u0137480/home/rods/dir_to_register:
  rods              0 default            6 2023-10-27.17:33 & File000.txt
        generic    /vault/home/rods/dir_to_register/File000.txt
  rods              0 default            6 2023-10-27.17:33 & File001.txt
        generic    /vault/home/rods/dir_to_register/File001.txt
``` 

Logs:
`irods stdout | {"log_category":"legacy","log_level":"error","log_message":"[-]\t/irods_source/server/api/src/rsFileReaddir.cpp:105:int _rsFileReaddir(rsComm_t *, fileReaddirInp_t *, rodsDirent_t **) :  status [Unknown iRODS error]  errno [Operation not permitted] -- message [fileReaddir failed for [/vault/home/rods/dir_to_register]]\n\t[-]\t/irods_source/server/drivers/src/fileDriver.cpp:443:irods::error fileReaddir(rsComm_t *, irods::first_class_object_ptr, struct rodsDirent **) :  status [Unknown iRODS error]  errno [Operation not permitted] -- message [failed to call 'readdir']\n\t\t[-]\t/irods_source/plugins/resources/unixfilesystem/src/unixfilesystem.cpp:1040:irods::error unix_file_readdir(irods::plugin_context &, struct rodsDirent **) :  status [Unknown iRODS error]  errno [Operation not permitted] -- message [End of directory list reached.]\n\n","request_api_name":"PHY_PATH_REG_AN","request_api_number":630,"request_api_version":"d","request_client_user":"rods","request_host":"10.0.2.100","request_proxy_user":"rods","request_release_version":"rods4.3.1","server_host":"dev-u0137480.irods.t.icts.kuleuven.be","server_pid":456935,"server_timestamp":"2023-10-27T15:33:58.217Z","server_type":"agent","server_zone":"u0137480"}
`

**Comment by korydraughn** [comment]
Based on the ils output, I'm guessing the data at rest is what you expect, correct?

Q. As for the logs, is there some policy being triggered on the server when you execute the registration?
Q. You mention the `prc`, how is that related to what you've presented?

**Comment by mstfdkmn** [comment]
Yes, the data at rest and in the "good - &" status is what is expected.

Q1: No, there is no rule active or enabled.
Q2: I wanted to say that I am getting the same error also when a collection is registered by the relevant method exists in PRC.

```
with iRODSSession(irods_env_file=env_file) as session:
    session.collections.register("/vault/home/rods/dir_to_register", "/u0137480/home/rods/dir_to_register")
```

**Comment by trel** [comment]
Are we running past the end of a list of filenames?

`End of directory list reached.`

**Comment by korydraughn** [comment]
I see an errno value of `Operation not permitted` and a reference to the following line.
https://github.com/irods/irods/blob/97eb33f130349db5e01a4b85e89dd1da81460345/plugins/resources/unixfilesystem/src/unixfilesystem.cpp#L1040

Assuming the errno is because of an OS API call, everything points to this line, which is just a few lines above the previous code snippet.
https://github.com/irods/irods/blob/97eb33f130349db5e01a4b85e89dd1da81460345/plugins/resources/unixfilesystem/src/unixfilesystem.cpp#L1028

Does iRODS have appropriate permissions to read that directory and the files in it?

**Comment by mstfdkmn** [comment]
Yes, as you might see in my first post, I gave irods all rights in the directory and the files in it. Also, if the user didnt have appropriate permissions, I guess it should not register it.

To reproduce this in a simple way, another thing is that a collection in irods can be unregistered and registered. Then the same error will be seen in logs. I mean the user has already rights required for that.

**Comment by korydraughn** [comment]
I was able to reproduce it.

We'll look into it.
